table/requirements.go

```diff
@@ -0,0 +1,212 @@
+package table
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+type Requirement interface {
+	Validate(Metadata) error
+}
+
+type baseRequirement struct {
+	Type string `json:"type"`
+}
+
+type AssertCreate struct {
+	baseRequirement
+}
```

Comment:
docstrings to explain what these are for? 

Also needs the Apache license header.